kyo-data/shared/src/main/scala/kyo/Text.scala

```diff
@@ -296,6 +607,24 @@ object Text:
             end substring
             override def toString: String = left.toString + right.toString
         end Concat
+
+        case class Reverse(payload: Text) extends Op:
+            def isEmpty: Boolean         = payload.isEmpty
+            def length: Int              = payload.length
+            def charAt(index: Int): Char = payload.charAt(length - 1 - index)
+            def substring(from: Int, until: Int): Text =
+                if from >= until then Text.empty
+                else if from == 0 && until == length then this
+                else Reverse(payload.substring(length - until, length - from))
+            override def toString: String =
```

Comment:
Should this be lazy val?

Reply:
I'm not sure, I think most instances of `Cut` will be transient won't ever produce a `String`. For example, `Parse` operates on `Text` without having to produce intermediate strings. The extra memory footprint for the two fields that `lazy val` desugars to doesn't seem worth it.

Reply:
Makes sense to me. I agree.